### ISSUE TITLE
GPC & DNT signal injector does not follow spec

### ISSUE THREAD

**Issue by eligrey** [EFForg/privacybadger#2788]
Your [Global Privacy Control / Do-Not-Track signal injector](https://github.com/EFForg/privacybadger/blob/d854233680592143cbf2bfc2c8da8bf9bf167ed5/src/js/contentscripts/dnt.js#L25-L44) does not follow the GPC or DNT specs. This is supposed to be exposed through getters on `Navigator.prototype`, not literal values set on `navigator`.

Consent managers may expect correct implementations to properly heed these signal. See the GPC implementation I added to OptMeOwt for reference: https://github.com/eligrey/optmeowt-browser-extension/blob/9be43907440708b742fb4b55b74ac94fb22fcf54/src/dom.js

#### What is your browser and browser version?

n/a

#### What is broken and where?

Privacy Badger's GPC & DNT signal injectors do not follow the relevant specifications, leading to incompatibility with some CMPs.

#### What is the "culprit" domain?

n/a

#### What is your debug output for this domain?

n/a

**Comment by ghostwords** [comment]
Hi and thanks for letting us know!

Could you explain what exactly the current implementation breaks? That is, which Consent Managers does the current implementation break, and what is broken?

Could you also point to which part of spec is violated by our current implementation?

Thanks again!

**Comment by ghostwords** [comment]
FYI, we used to set GPC on the Navigator prototype via a getter, but we [updated our approach](https://github.com/EFForg/privacybadger/pull/2722/files?w=1) in response to incompatibility with other extensions (Duck Duck Go specifically) as part of #2722. See the discussion in beginning in https://github.com/globalprivacycontrol/gpc-spec/issues/11#issuecomment-732450507 for background.

**Comment by eligrey** [comment]
> Could you explain what exactly the current implementation breaks? That is, which Consent Managers does the current implementation break, and what is broken?

Transcend Consent Manager explicitly reads `Object.getOwnPropertyDescriptor(Navigator.prototype, 'globalPrivacyControl')?.get?.call(navigator)` (same for `navigator.doNotTrack`) due to our strict standards for reference security to prevent prototype pollution vulnerabilities and manipulation by hostile ads. Read more: https://transcend.io/blog/defeating-cookie-banners).

> Could you also point to which part of spec is violated by our current implementation?

Both specs define these as exposed on `Navigator` (class), and not hardcoded on `navigator` (an specific instance of `Navigator`):

- Global Privacy Control: https://globalprivacycontrol.github.io/gpc-spec/#javascript-property-to-detect-preference
- Do Not Track: https://www.w3.org/TR/tracking-dnt/#js-dom

> FYI, we used to set GPC on the Navigator prototype via a getter, but we updated our approach in response to incompatibility with other extensions (DuckDuckGo Privacy Essentials specifically) as part of #2722

You got the following error because you implemented the property descriptor as non-configurable. It's supposed to be `configurable: true`, but your previous code had an implicit `configurable: false` by not including this in the property descriptor.

```
typeError: can't redefine non-configurable property "globalPrivacyControl"
```

You should not run into this issue if you revert to that implementation with `configurable: true` and `enumerable: true`. That should have been the fix to the DDG conflict issue.

**Comment by ghostwords** [comment]
OK, thank you for explaining! Would you be interested in opening a pull request?

**Comment by eligrey** [comment]
Yeah, I can open up a PR later today or tomorrow 👍. I'll request your review once it's up

**Comment by eligrey** [comment]
@ghostwords here's a PR for you to check out: https://github.com/EFForg/privacybadger/pull/2790

**Comment by michaelfarrell76** [comment]
hey @ghostwords can we merge this pr?
